### PR TITLE
finish tests and endpoints for get comments by article id

### PR DESCRIPTION
**Pull request by DominiqueCoding**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -142,4 +142,44 @@ describe("GET /api/articles",()=>{
...
     })
+})
+
+describe("GET /api/articles/:article_id/comments",()=>{
+    it("200:should return with array of comments that match article id",()=>{
+        return request(app).get("/api/articles/1/comments").expect(200)
+        .then(({body})=>{
+
+            const commentArr = body
+
```
Comment (nc-hev-mag) [inline __tests__/app.test.js]: add a check for the length of the array here too since we are asserting inside a forEach

File: endpoints.json [hunk 2]
```diff
@@ -43,5 +43,29 @@
         }
       ]
     }
+  },
+  "GET /api/articles/:article_id/comments":{
```
Comment (nc-hev-mag) [inline endpoints.json:47]: nice!

File: utils.js [hunk 3]
```diff
@@ -1,14 +1,139 @@
...
-    return db.query(query,[id])
-    .then(({rows}) =>{
-        if(!rows.length){
-            Promise.reject({status:404,msg:"not found"})
-        }
-        // return rows
-    })
-}
\ No newline at end of file
+[
```
Comment (nc-hev-mag) [inline utils.js]: try not to commit big chunks of commented out code like this

**Review by nc-hev-mag — CHANGES_REQUESTED** [review]
Hey Dominique! Great tests here so far - just need a couple more errors covered

- status 404: valid article_id but does not exist i.e. 9999
- status 400: invalid article_id i.e. notanumber

Please resubmit once added 😄

File: utils.js (hunk 3 above)
Comment (DominiqueCoding) [reply utils.js]: i was trying something and forgot about it when i didn't need it anymore, will change 👍

File: __tests__/app.test.js [hunk 4]
```diff
@@ -145,40 +145,49 @@ describe("GET /api/articles",()=>{
...
-                    article_id: expect.any(Number)
+                    article_id: 1
                 })
             })
 
      
         })
     })
 
-    it("200:should return with an empty array for articles with no comments",()=>{
```
Comment (saimakhanom) [inline __tests__/app.test.js]: This was a good and valid test - why did you get rid of it?

**Comment by saimakhanom** [comment]
Looks good to merge, do bring back the empty comment array test first though - it's a good one!

**Pull request merged**